src/Bindings/ManualBindings.h

```diff
@@ -51,8 +51,26 @@ class cManualBindings
 	// Helper functions:
 	static cPluginLua * GetLuaPlugin(lua_State * L);
 	static int tolua_do_error(lua_State * L, const char * a_pMsg, tolua_Error * a_pToLuaError);
-	static int lua_do_error(lua_State * L, const char * a_pFormat, fmt::ArgList a_ArgList);
-	FMT_VARIADIC(static int, lua_do_error, lua_State *, const char *)
+
+	template <typename... Args>
+	static int lua_do_error(lua_State * L, const char * a_pFormat, const Args & ... a_ArgList)
```

Comment:
We used `fmt::ArgList` for a reason, this shouldn't be a template. The new equivalent appears to be `fmt::format_args`.

Reply:
Also, only `fmt/core.h` should be included globally.

Reply:
Thanks, but I'm unsure how to use this, it seems the FMT_VARIADIC macro was removed in favour of templates? What was the reason to use ArgList?

Reply:
And besides, it is known that templates = ***SPEED***

Reply:
`fmt` is designed to minimised the code templated on the input argument types in order to reduce binary size. In our case it also means we don't need to include all of `fmt` everywhere, just `fmt/core.h`.

```cpp
void Log(const char * a_Format, fmt::format_args);

template <typename Args...>
void Log(const char * a_Format, const Args & ... args)
{
	Log(a_Format, fmt::make_format_args(args...));
}
```

Reply:
I couldn't get this compiling, it claims that it can't find a formatter for Vector3. Honestly I have no idea why.

I cannot repreoduce binary size claims, for me:

Windoze
master MSVC Release: 7.63MB
this PR MSVC Release: 5.37MB

Leenucks
master Ubuntu Bionic Release: 11MB
this PR Ubuntu Bionic Release: 8.2MB

This seems to suggest that the changes are worth it.

Reply:
Okay, I think this is what we want:
https://gcc.godbolt.org/z/AAzUog

Turns out that the `fmt::format_args` version can't have the same name as the template. Renaming it `vLog` did the trick.

Also note that the `printf`-like functions now have their own args class. `fmt::printf_args` and the corresponding `fmt::make_printf_args` which requires `fmt/printf.h` to be included.